File: LeetCode/1816_Truncate_Sentence/solution.js
/**
 * @param {string} s
 * @param {number} k
 * @return {string}
 */
var truncateSentence = function(s, k) {
    let m=s.trim();
    let n=""
    let temp=""
    let count=0;
    for(let i=0;i<m.length;i++ ){
        if(s[i]!=" ")temp+=m[i];
        if(s[i]==" "&&s[i-1]!=" "){
            n+=temp;
            temp=" "
            count++;
            if(count==k) break;
        }
        else if(s[i]!=" " && i==m.length-1) {
            n+=temp;
            temp=" "
            count++;
        }
    }
    return n;
};

File: LeetCode/1816_Truncate_Sentence/solution.cpp
class Solution {
public:
    string truncateSentence(string s, int k) {
        // let m = s.trim();
        string n = ""; 
        string temp = ""; 
        int count = 0;
        for (int i = 0; i < s.size(); i++) {
            if (s[i] != ' ')
                temp += s[i];
            if (s[i] == ' ' && s[i - 1] != ' ') {
                n += temp;
                temp = ' '; 
                count++;
                if (count == k)
                    break;
            } else if (s[i] != ' ' && i == s.size() - 1) {
                n += temp;
                temp = ' ';
                count++;
            }
        }
        return n;
    }
};